src/_articles/libraries/dart-io.md

```diff
@@ -0,0 +1,386 @@
+Exiting the `await for` statement — using `break`, `return`, or an uncaught exception —
+cancels the subscription.
+
+`Stream<List<int>>` is used in multiple places in dart:io:
+when working with stdin, files, sockets, HTTP connections, and so on.
+Similarly, [IOSink]({{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/IOSink-class.html) objects
+are used to stream data to
+stdout, files, sockets, HTTP connections, and so on.
+
+## Interacting with processes
+
+For the simple case, use
+[Process.run()]({{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/Process/run.html)
+to run a process
+and collect its output. Use `run()` when you don't
+need interactive control over the process.
+
+<?code-excerpt "misc/lib/articles/io/io_process_test.dart"?>
+{% prettify dart %}
+import 'dart:async' show Future;
+import 'dart:io';
+
+Future<void> main() async {
+  // List all files in the current directory,
+  // in UNIX-like operating systems.
+  ProcessResult results = await Process.run('ls', ['-l']);
+  print(results.stdout);
+}
+{% endprettify %}
+
+You can also start a process by creating a
+[Process]({{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/Process-class.html) object
+using
+[Process.start().]({{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/Process/start.html)
+
+Once you have a Process object you can interact with the process
+by writing data to its stdin sink,
+reading data from its stderr and stdout streams,
+and killing the process.
+When the process exits, the `exitCode` future completes with
+the exit code of the process.
+
+The following example runs `ls -l` in a separate process
+and prints the output and the exit code for the process to stdout.
+Since we are interested in getting lines,
+we use a
+[Utf8Decoder]({{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-convert/Utf8Decoder-class.html)
+(which decodes chunks of bytes into strings) followed by a
+[LineSplitter]({{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-convert/LineSplitter-class.html)
+(which splits the strings at line boundaries).
+
+<?code-excerpt "misc/lib/articles/io/io_process_transform_test.dart"?>
+{% prettify dart %}
+import 'dart:async' show Future;
+import 'dart:convert';
+import 'dart:io';
+
+Future<void> main() async {
+  Process process = await Process.start('ls', ['-l']);
+  var lineStream =
+      process.stdout.transform(Utf8Decoder()).transform(LineSplitter());
+  await for (var line in lineStream) {
+    print(line);
+  }
+
+  await process.stderr.drain();
+  print('exit code: ${await process.exitCode}');
+}
+{% endprettify %}
+
+Notice that `exitCode` can complete before all of the lines of output
+have been processed. Also note
+that we do not explicitly close the process. In order to
+not leak resources we have to drain both the stderr and the stdout
+streams. To do that we set a listener (using `await for`)
+to drain the stderr stream.
```

Comment:
For `stderr` this code example does not use `await for`. Maybe rephrase this to something like this:

.. we have to listen to both streams, stderr and stdout. We use `await for` for listening to the stdout of the subprocess whereas we use `stderr.drain()`, which will listen to a stream and discard it.